proposed/caller-unsafe.md

```diff
@@ -0,0 +1,60 @@
+
+# Annotating unsafe code with RequiresUnsafe
```

Comment:
@jaredpar started writing a draft for this as well some time ago: https://github.com/jaredpar/csharplang/tree/unsafe . 

(It proposes to reuse the `unsafe` keyword that I think is much cleaner design.)